### ISSUE TITLE
Model converted to ONNX give different results 

### ISSUE THREAD

**Issue by mlaszko** [onnx/tensorflow-onnx#1597]
**Describe the bug**
Model converted to ONNX(run with onnxruntime)) give different results that original tf model(run with tensorflow keras).

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 20.04.2 LTS
- Tensorflow Version: 2.5.0
- Python version: 3.8.10 

**To Reproduce**
[Saved model](https://drive.google.com/file/d/1odRjDT5kT9LQduoiioeWKKREsQf7yMhP/view?usp=sharing)
`python -m tf2onnx.convert --saved-model model --opset 14 --output output.onnx --tag serve --signature_def serving_default`

```
import onnxruntime as rt
from onnxruntime.datasets import get_example
import cv2
import numpy as np

img = cv2.imread("file.jpg")
img = img.astype(np.float32)

model = get_example("output.onnx")
sess = rt.InferenceSession(model, providers=["CPUExecutionProvider"])
input_name = sess.get_inputs()[0].name
output_name = sess.get_outputs()[0].name
result = sess.run([output_name], {input_name: img})
```

**Comment by TomWildenhain-Microsoft** [comment]
Why are you using `get_example`?  You should be able to just do:
`sess = rt.InferenceSession("output.onnx", providers=["CPUExecutionProvider"])`

I think you might be comparing against an example model, not the converted model.

**Comment by mlaszko** [comment]
I removed get_example, but result haven't changed.

**Comment by TomWildenhain-Microsoft** [comment]
I ran your script on the converted result, but I got that the input shape doesn't match the model.  Can you give me a copy of the input you are using?

**Comment by mlaszko** [comment]
Model shape is [-1, 32, -1, 3]. Sample image:
![10_32](https://user-images.githubusercontent.com/1485342/124720720-ba1b0f00-df08-11eb-86a0-ce1b9a80b677.jpg)

**Comment by TomWildenhain-Microsoft** [comment]
I still got an incorrect shape but when I added in a 1 for the batch dim I was able to run in tf and onnx and the results match.

```
import onnxruntime as rt
import cv2
import numpy as np
import tensorflow as tf

img = cv2.imread("img.jpg")
img = img.astype(np.float32)
img = img.reshape([1] + list(img.shape))

sess = rt.InferenceSession("output.onnx", providers=["CPUExecutionProvider"])
input_name = sess.get_inputs()[0].name
output_name = sess.get_outputs()[0].name
result = sess.run([output_name], {input_name: img})
print("ONNX")
print(result[0])

model = tf.saved_model.load("model")
res_tf = model(img)
print("Tensorflow")
print(res_tf)

np.testing.assert_allclose(res_tf, result[0], rtol=1e-5, atol=1e-5)
print("Results match")
```

Output:
```
ONNX
[[[ -0.9718218   -4.5728645   14.452157    -4.776663    -3.085457 
    -7.9910655   -1.6243386   -3.7887735   -0.8151489   -1.7057282
    -2.7424536   -7.4520254   -8.166557    -7.748675    -7.3275027
    -1.3288321 ]
  [ -4.253899    -6.9572654   13.5326805   -5.7516313   -3.8529594
    -7.807081    -3.775939    -5.353466    -2.387596    -2.128575 
    -0.6421658   -8.321213    -8.979596    -8.447499    -8.09948  
     5.2182865 ]
  [ -8.195577    -4.4302115    2.523629    -4.76802     -5.5698137
    -5.4468603   -5.9887433   -6.1446633   -4.9890637   -2.3037457
     0.6628251   -6.777722    -9.100241    -9.191741    -8.2653675
    16.238865  ]
  [-10.396589     3.183912    -4.55783     -2.2898536   -5.819543 
    -4.1167345   -6.1484513   -5.3076177   -5.8298707   -1.2814585
    -0.5806415   -5.683783    -8.511952    -8.759041    -7.8003597
    17.428997  ]
  [-10.262247    11.264116    -5.795656     0.500815    -4.832449
    -2.5888293   -5.4112177   -4.071131    -4.830731     1.8541889
     0.6139955   -5.2213016   -6.106772    -6.225189    -6.1063395
     8.918736  ]
  [ -5.1736665    7.783113    -3.9496424    2.2280197   -2.9905748
    -0.21877152  -3.0606768   -2.8755305   -0.6025062    7.2334476
     4.6740575   -4.7658477   -4.521438    -3.4980931   -3.583794
    -3.3653593 ]]]
2021-07-07 01:20:38.797637: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2021-07-07 01:20:38.797915: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed call to cuInit: UNKNOWN ERROR (303)
2021-07-07 01:20:38.805447: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: DESKTOP-B2J6DTH
2021-07-07 01:20:38.806294: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: DESKTOP-B2J6DTH
2021-07-07 01:20:38.806817: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural 
Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-07-07 01:20:45.564172: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled 
(registered 2)
Tensorflow
tf.Tensor(
[[[ -0.9718214   -4.572865    14.452158    -4.7766633   -3.085457 
    -7.991065    -1.6243382   -3.7887733   -0.81514937  -1.7057284
    -2.7424538   -7.4520245   -8.166558    -7.7486753   -7.327503 
    -1.3288323 ]
  [ -4.2539005   -6.9572663   13.53268     -5.751631    -3.8529596
    -7.807081    -3.7759397   -5.3534656   -2.3875957   -2.1285746
    -0.64216596  -8.321214    -8.979595    -8.447499    -8.099479 
     5.218287  ]
  [ -8.195578    -4.430212     2.5236285   -4.7680182   -5.569813 
    -5.4468594   -5.9887433   -6.1446643   -4.9890647   -2.3037457
     0.6628251   -6.777721    -9.100241    -9.191741    -8.265368 
    16.238865  ]
  [-10.396589     3.1839116   -4.5578294   -2.2898524   -5.819542 
    -4.116735    -6.148452    -5.307618    -5.8298717   -1.2814585
    -0.58064187  -5.6837826   -8.511953    -8.75904     -7.8003597
    17.428997  ]
  [-10.262247    11.264116    -5.795656     0.5008156   -4.832448
    -2.5888278   -5.411217    -4.0711308   -4.8307304    1.8541884
     0.6139954   -5.221301    -6.106772    -6.2251883   -6.106339
     8.918737  ]
  [ -5.173667     7.783113    -3.9496427    2.2280204   -2.9905741
    -0.21877116  -3.0606775   -2.8755314   -0.6025056    7.2334466
     4.6740565   -4.7658467   -4.521438    -3.498093    -3.5837946
    -3.3653588 ]]], shape=(1, 6, 16), dtype=float32)
Results match
```

If you are still getting a mismatch, can you please provide the code you are using to run the tensorflow model?  Also please upload the onnx model you are using.

**Comment by fatcat-z** [comment]
No response from user yet, close it.
